clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -259,18 +271,21 @@ public void completed(
           long offset,
           String message,
           Optional<LeaderFollowerStateType> leaderState) {
-        if (isTopicPartitionIngesting(kafkaTopic, partition)) {
-          VeniceStoreVersionConfig config = configLoader.getStoreConfig(kafkaTopic);
-          config.setRestoreDataPartitions(false);
-          config.setRestoreMetadataPartition(false);
-          // Start partition consumption locally.
-          startConsumption(config, partition, leaderState);
-        } else {
-          LOGGER.error(
-              "Partition: {} of topic: {} is not assigned to this host, will not resume the ingestion on main process.",
-              partition,
-              kafkaTopic);
-        }
+        // Use thread pool to handle the completion reporting to make sure it is not blocking the report.
+        getCompletionHandlingExecutor().submit(() -> {
+          if (isTopicPartitionIngesting(kafkaTopic, partition)) {
```

Comment:
Shouldn't this (or some other function...) return false if the partition was dropped? So why make all of this async?

Reply:
I think the dropping partition action and removing store version in store repo is different operations. This method only check the II data structure maintained state. The standby->offline and offline->drop message will come later when we hit this part.

But you are right, actually we don't need async on all part, maybe just the internal start consumption. I will update the code.